### ISSUE TITLE
IndexedTxGraph::batch_insert_* methods are overwraught

### ISSUE THREAD

**Issue by LLFourn** [bitcoindevkit/bdk#1260]
We have:

- `batch_insert_unconfirmed`
- `batch_insert_relevant_unconfirmed`
- `batch_insert_relevant`

We only need `batch_insert_relevant` which should have the API:

```rust
    pub fn batch_insert_relevant<'t>(
        &mut self,
        txs: impl IntoIterator<Item = &'t Transaction>,
    ) -> ChangeSet<A, I::ChangeSet> {
```

Whether they are confirmed or unconfirmed (have only anchors or last seen) is not important for doing this job. The caller can add anchors and last seen with the methods available for that. Also the documentation should make it clear why this method exists in the first place: because relevancy cannot be determined until all of the transactions have been first scanned by the indexer.

Also there is a `batch_insert_unconfirmed` in `TxGraph` that doesn't need to exist (can be achieved by a for loop).

**Comment by LLFourn** [comment]
It looks like this was already sort of noted but we forgot to fix it: https://github.com/bitcoindevkit/bdk/pull/1041#discussion_r1347427162

**Comment by evanlinjin** [comment]
The caller will have to iterate over the changeset returned from `batch_insert_relevant` to see if `last_seen` and/or anchors are worth adding. Wouldn't that be cumbersome?

**Comment by LLFourn** [comment]
Good point. I hadn't considered that. I think we can at least remove `batch_insert_unconfirmed` from `TxGraph` and `IndexedTxGraph`.
